### PR TITLE
[Heartbeat] Bump Node to 18.18

### PR DESCRIPTION
**Pull request by emilioalvap**
## Why is it important?

Bump NodeJS version bundled with Heartbeat to 18.18.2.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
- [x] I have added an integration test or an E2E test

## How to test this PR locally

1. Build mage package locally.
2. Run a container with the generated image.
3. Execute the following command and test version matches `18.18.2`:
```
docker run --rm --entrypoint /bin/bash docker.elastic.co/beats/elastic-agent-complete:<version> -c "node --version"
v18.17.1
```

### CODE REVIEW

File: changelog/fragments/1698247934-hb-bump-node.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+# - other: does not fit into any of the other categories
+kind: bug-fix
+
+# Change summary; a 80ish characters long description of the change.
+summary: hb-bump-node-18.17
+
+# Long description; in case the summary is not enough to describe the change
+# this field accommodate a description without length limits.
+# NOTE: This field will be rendered only for breaking-change and known-issue kinds at the moment.
+description: Bump NodeJS version bundled with Heartbeat to v18.17.1.
```
Comment (emilioalvap) [inline changelog/fragments/1698247934-hb-bump-node.yaml:19]:
```suggestion
description: Bump NodeJS version bundled with Heartbeat to v18.18.2.
```

File: dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl [hunk 2]
```diff
@@ -170,7 +170,7 @@ RUN echo \
 ENV ELASTIC_SYNTHETICS_CAPABLE=true
 ENV ELASTIC_AGENT_COMPLETE=true
 ENV TZ=UTC
-ENV NODE_VERSION=18.16.0
+ENV NODE_VERSION=18.17.1
```
Comment (emilioalvap) [inline dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl:173]:
```suggestion
ENV NODE_VERSION=18.18.2
```

File: changelog/fragments/1698247934-hb-bump-node.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+# - breaking-change: a change to previously-documented behavior
+# - deprecation: functionality that is being removed in a later release
+# - bug-fix: fixes a problem in a previous version
+# - enhancement: extends functionality but does not break or fix existing behavior
+# - feature: new functionality
+# - known-issue: problems that we are aware of in a given version
+# - security: impacts on the security of a product or a user’s deployment.
+# - upgrade: important information for someone upgrading from a prior version
+# - other: does not fit into any of the other categories
+kind: bug-fix
```
Comment (pierrehilbert) [inline changelog/fragments/1698247934-hb-bump-node.yaml:11]: This is not a bug-fix, other than that, LGTM.

**Review by pierrehilbert — APPROVED** [review]
LGTM

**Review by vigneshshanmugam — APPROVED** [review]

File: changelog/fragments/1698247934-hb-bump-node.yaml (hunk 3 above)
Comment (emilioalvap) [inline changelog/fragments/1698247934-hb-bump-node.yaml:11]:
```suggestion
kind: other
```

**Pull request merged**